keras_cv/layers/regularization/dropblock_2d.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from keras.engine.base_layer import BaseRandomLayer
```

Comment:
I believe we only want to rely on symbols exported under the `tf.keras` namespace which is actually distinct from Keras.

I will add an export to tf.keras.__internal__.BaseRandomLayer for you to use.  You will need to use Keras nightly once it is merged.

https://github.com/keras-team/keras/pull/16203

Reply:
Thanks, I can confirm that I can import `from tensorflow.keras.__internal__.layers import BaseRandomLayer` in `tf-nightly`. 

Do you know if the `setup.py` in this repository will also be changed to use `tf-nightly` instead of stable TF? If this is a problem I can just use the `tf.random.Generator` directly in this layer (and inherit from `Layer`).

Reply:
I can do that now actually.

Reply:
Done.

Reply:
Thanks. Updated the layer to use `BaseRandomLayer`.